src/ipc/ipc-helper.c

```diff
@@ -187,7 +187,12 @@ int comp_buffer_connect(struct comp_dev *comp, uint32_t comp_core,
 	if (buffer->core != comp_core) {
 
 		/* set the buffer as a coherent object */
+#ifdef __ZEPHYR__
+		/* under Zephyr buffers are only used in threads */
+		coherent_shared_thread(buffer, c);
+#else
 		coherent_shared(buffer, c);
+#endif
```

Comment:
Can your next PR abstract this at the coherent API level (i.e. we have no client code that does #if Zephyr). Thanks.

Reply:
@lgirdwood Actually yes, you're right, my mistake, let me fix this